src/Server/Remote/NativeSsh.php

```diff
@@ -244,32 +263,118 @@ protected function getMultiplexingSshOptions()
      * but after mux is created with exec() then Symfony Process
      * can work with it.
      */
-    public function initMultiplexing()
+    private function initMultiplexing()
     {
         $serverConfig = $this->getConfiguration();
-        $username = $serverConfig->getUser() ? $serverConfig->getUser() . '@' : null;
-        $hostname = $serverConfig->getHost();
 
         $sshOptions = [];
+
         if ($serverConfig->getConfigFile()) {
-            $sshOptions[] = '-F ' . escapeshellarg($serverConfig->getConfigFile());
+            $sshOptions = array_merge_recursive($sshOptions, ['-F' => $serverConfig->getConfigFile()]);
         }
+
         if ($serverConfig->getPort()) {
-            $sshOptions[] = '-p ' . escapeshellarg($serverConfig->getPort());
+            $sshOptions = array_merge_recursive($sshOptions, ['-p' => $serverConfig->getPort()]);
         }
+
         if ($serverConfig->getPrivateKey()) {
-            $sshOptions[] = '-i ' . escapeshellarg($serverConfig->getPrivateKey());
+            $sshOptions = array_merge_recursive($sshOptions, ['-i' => $serverConfig->getPrivateKey()]);
         } elseif ($serverConfig->getPemFile()) {
-            $sshOptions[] = '-i ' . escapeshellarg($serverConfig->getPemFile());
+            $sshOptions = array_merge_recursive($sshOptions, ['-i' => $serverConfig->getPemFile()]);
         }
-        $sshOptions = array_merge($sshOptions, $this->getMultiplexingSshOptions());
 
-        exec('ssh ' . implode(' ', $sshOptions) . ' -O check -S ' . $this->getConnectionHash() . ' ' . escapeshellarg($username . $hostname) . ' 2>&1', $checkifMuxActive);
-        if (!preg_match('/Master running/', $checkifMuxActive[0])) {
+        $sshOptions = array_merge_recursive($sshOptions, $this->getMultiplexingSshOptions());
+
+        $sshOptions = array_merge_recursive($sshOptions, [
+            $serverConfig->getUserAndHost() => null,
+        ]);
+
+        exec('ssh -O check -S ' . $this->getConnectionHash() . ' ' . $this->buildCommandLineOptions($sshOptions, true) . ' 2>&1', $checkIfMuxActive);
+        if (!preg_match('/Master running/', $checkIfMuxActive[0])) {
             if (\Deployer\isVerbose()) {
                 \Deployer\writeln('  SSH multiplexing initialization');
             }
-            exec('ssh ' . implode(' ', $sshOptions) . ' ' . escapeshellarg($username . $hostname) . "  'echo \"SSH multiplexing initialization\"' ");
+            exec('ssh ' . $this->buildCommandLineOptions($sshOptions, true) . "  'echo \"SSH multiplexing initialization\"' ");
```

Comment:
Actually I didn't test it after options refactoring because I'm not sure how to do that correctly.